src/video.cpp

```diff
@@ -377,16 +360,19 @@ void init_window(bool hidden)
 		window_flags |= SDL_WINDOW_HIDDEN;
 	}
 
-	uint32_t renderer_flags = SDL_RENDERER_ACCELERATED | SDL_RENDERER_TARGETTEXTURE;
+	// Initialize window
+	window.reset(new sdl::window("", w, h, window_flags));
+	if(prefs::get().fullscreen()) {
+		SDL_DisplayMode mode;
+		mode.format = SDL_PIXELFORMAT_UNKNOWN;
+		mode.w = w;
+		mode.h = h;
+		mode.refresh_rate = 0;
+		mode.internal = 0;
 
-	if(prefs::get().vsync()) {
-		LOG_DP << "VSYNC on";
-		renderer_flags |= SDL_RENDERER_PRESENTVSYNC;
+		SDL_SetWindowFullscreenMode(*window, &mode);
```

Comment:
Creating the display mode data manually seems wrong. https://wiki.libsdl.org/SDL3/SDL_SetWindowFullscreenMode says we could just pass null for a borderless fullscreen window (which AFAIK is what we want anyway, since we used `SDL_WINDOW_FULLSCREEN_DESKTOP` before.

Reply:
Doing it that way puts the Wesnoth window on the other display instead of the one it was on before being made fullscreen and also does something (scales things up?) that seems to cause the window size SDL creates to be larger than the actual display (aka there's a large amount of the window cut off on the edges).

Reply:
Hmm... I wonder if that's a bug. Or maybe some issue with our manual pixel scaling and the new HDPI handling.